### PR TITLE
Docs UI update

### PR DESCRIPTION
**Pull request by ZYJLiu**
1. Add Tabs/Tab component
CleanShot 2024-07-30 at 12 25 18

Example usage on developer-content md files:
```md
<!-- prettier-ignore -->
<Tabs groupId="language" items={['Rust', 'Typescript', 'Java']}>
<Tab value="Rust">Rust 1</Tab>
<Tab value="Typescript">Typescript 1</Tab>
<Tab value="Java">Java</Tab>
</Tabs>

<!-- prettier-ignore -->
<Tabs groupId="language" items={['Rust', 'Typescript', 'Go']}>
<Tab value="Rust">Rust 2</Tab>
<Tab value="Typescript">Typescript 2</Tab>
<Tab value="Go">Go</Tab>
</Tabs>

<!-- prettier-ignore -->
<Tabs groupId="package-managers" items={['pnpm', 'npm', 'yarn']}>
<Tab value="pnpm">pnpm 1</Tab> 
<Tab value="npm">npm 1</Tab>
<Tab value="yarn">yarn 1</Tab> 
</Tabs>

<!-- prettier-ignore -->
<Tabs groupId="package-managers" items={['pnpm', 'npm', 'yarn']}>
<Tab value="pnpm">pnpm 2</Tab> 
<Tab value="npm">npm 2</Tab>
<Tab value="yarn">yarn 2</Tab> 
</Tabs>
```
2. Add Accordion/AccordionItem component

CleanShot 2024-07-30 at 12 27 32

Example usage on developer-content md files:

````md
  <Accordion>
  <AccordionItem title="Hello">
  
  ```ts
  console.log("hello");
  ```
  
  </AccordionItem>
  <AccordionItem title="World">
  
  ```rs
  println!("world");
  ```
  
  </AccordionItem>
  </Accordion>
````

3. Update docs codeblock UI with svgs icons in header
CleanShot 2024-07-30 at 12 20 40@2x

4. Add lightmode color to docs components/codeblock
CleanShot 2024-07-30 at 12 21 46@2x
CleanShot 2024-07-30 at 12 28 20@2x
CleanShot 2024-07-30 at 12 29 56@2x

5. Update docs scrollbar to only appear on hover
CleanShot 2024-07-30 at 12 31 06

6. Fix doc table of content bug where "SOL" gets converted to "SO"
7. Update Table of Contents "Edit Page" and "Scroll to Top" buttons 
CleanShot 2024-07-31 at 10 20 03

8. Add Stack Exchange to docs secondary nav
CleanShot 2024-07-31 at 10 22 19@2x

### CODE REVIEW

File: src/utils/contentApi.ts [hunk 1]
```diff
@@ -353,7 +353,7 @@ export default class ContentApi {
 
         // we remove any special characters from the end of the label
         // to make it look cleaner in the UI
-        label = label.split(/[\v{L|N}]*$/)[0];
+        label = label.replace(/[^a-zA-Z0-9]+$/, "");
```
Comment (nickfrosty) [inline src/utils/contentApi.ts]: updating this regex to your suggesest change will break all translated content within the table of contents. and maybe prevent the site from building.

we need to find a different solution to solve the "trailing L" bug we are seeing

File: src/utils/contentApi.ts [hunk 2]
```diff
@@ -388,7 +388,7 @@ export default class ContentApi {
         .trim()
         .replace(/\s+/g, "-")
         .replace(/\./g, "")
-        .replace(/[[\v{L|N}]*]+/g, "-")
+        .replace(/[^a-z0-9]+/g, "-")
```
Comment (nickfrosty) [inline src/utils/contentApi.ts]: same comment as above

File: src/components/developers/DevelopersNav/DevelopersNav.jsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ import CookbookIcon from "../../../../public/src/img/developers/cookbook.inline.
 import GuidesIcon from "../../../../public/src/img/developers/guides.inline.svg";
 import TerminologyIcon from "../../../../public/src/img/developers/terminology.inline.svg";
 import CoursesIcon from "../../../../public/src/img/developers/courses.inline.svg";
+import StackExchangeIcon from "../../../../assets/developers/stackexchange.inline.svg";
```
Comment (nickfrosty) [inline src/components/developers/DevelopersNav/DevelopersNav.jsx]: can we cleanup these relative paths by using `@@/public/**`? either in this PR or a different one?

**Review by nickfrosty — CHANGES_REQUESTED** [review]
great work here John! the new components will be a great addition to the site. better UX all around! :)

I just have the one concern about the regex fix for other languages. we need to find a different solution

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Please add the usage guidelines for the new components (in the text of this PR) into `CONTRIBUTING.md`

File: src/components/developers/DevelopersNav/DevelopersNav.jsx (hunk 3 above)
Comment (ZYJLiu) [reply src/components/developers/DevelopersNav/DevelopersNav.jsx]: updated

**Comment by ZYJLiu** [comment]
> Please add the usage guidelines for the new components (you've already added them in the text of this PR) into `CONTRIBUTING.md`

will add to developer-content, since that is where the components are used

**Review by nickfrosty — APPROVED** [review]
Great work on this John. I think people are going to really like these new UX improvements

**Comment by nickfrosty** [comment]
> Please add the usage guidelines for the new components (you've already added them in the text of this PR) into `CONTRIBUTING.md`

added tracking issue on dev content: https://github.com/solana-foundation/developer-content/issues/288

File: src/utils/contentApi.ts [hunk 4]
```diff
@@ -353,7 +353,7 @@ export default class ContentApi {
 
         // we remove any special characters from the end of the label
         // to make it look cleaner in the UI
-        label = label.replace(/[^a-zA-Z0-9]+$/, "");
+        label = label.replace(/[^\p{L}\p{N}]+$/u, "");
```
Comment (mikemaccana) [inline src/utils/contentApi.ts:357]: BTW @ZYJLiu you can use `\p{Letter}` and `\p{Number}` to make the unicode bits more obvious. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escape

**Pull request merged**